chia/wallet/trade_manager.py

```diff
@@ -484,10 +488,12 @@ async def _create_offer_for_ids(
                     memos: List[bytes] = []
                     if isinstance(id, int):
                         wallet_id = uint32(id)
-                        wallet = self.wallet_state_manager.wallets[wallet_id]
+                        wallet = self.wallet_state_manager.wallets.get(wallet_id)
+                        assert isinstance(wallet, (CATWallet, Wallet))
                         p2_ph: bytes32 = await wallet.get_puzzle_hash(new=not tx_config.reuse_puzhash)
                         if wallet.type() != WalletType.STANDARD_WALLET:
                             if callable(getattr(wallet, "get_asset_id", None)):  # ATTENTION: new wallets
+                                assert isinstance(wallet, CATWallet)
```

Comment:
This should be present on all wallets that are not `Wallet` type too (or at least that was the intention).  If this is not the case, then this is fine, but if it is the case and the `not isinstance` trick from above works, maybe we should switch this out with that and maybe even replace `if wallet.type() ...` with that instead.